services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtDataPublisher.java

```diff
@@ -153,13 +159,22 @@ private void flushBuffer() {
                 }
 
                 DataEvent dataEvent = (DataEvent) event;
+
+                int size = dataEvent.data().remaining();
```

Comment:
Is it possible for dataEvent to have null/no data ?

Reply:
It should not be null, but I'll add a validation in the spirit of defensive programming.